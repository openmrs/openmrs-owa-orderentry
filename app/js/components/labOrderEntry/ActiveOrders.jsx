import React from 'react';
import PropTypes from 'prop-types';

const ActiveOrders = ({ labOrderData, tests }) => (
  <div>
    <table className="table bordered mw-958-px">
      <thead>
        <tr>
          <th>Name</th>
          <th>tests</th>
          <th>status</th>
        </tr>
      </thead>
      <tbody>
        <tr key={labOrderData.uuid}>
          <td>{labOrderData.display}</td>
          <td>{tests.join(', ')}</td>
          <td>awaiting-results</td>
        </tr>
      </tbody>
    </table>
  </div>
);

export default ActiveOrders;

ActiveOrders.defaultProps = {
  labOrderData: {
    uuid: '',
    display: '',
  },
  tests: [''],
};

ActiveOrders.propTypes = {
  labOrderData: PropTypes.shape({
    uuid: PropTypes.string,
    display: PropTypes.string,
  }),
  tests: PropTypes.arrayOf(PropTypes.string),
};
