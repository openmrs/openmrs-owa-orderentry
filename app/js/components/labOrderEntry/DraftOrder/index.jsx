import React from 'react';
import PropTypes from 'prop-types';

import DraftList from './DraftList';

export class LabDraftOrder extends React.Component {
  render() {
    const { draftLabOrders } = this.props;
    const numberOfDraftOrders = draftLabOrders.length;
    const isDisabled = !numberOfDraftOrders;
    return (
      <div className="draft-spacing draft-lab-layout">
        <h5 className="h5-draft-header">
          Unsaved Draft Orders ({numberOfDraftOrders})
        </h5>
        <div className="table-container">
          <ul className="draft-list-container">
            <DraftList
              draftOrders={draftLabOrders}
            />
          </ul>
        </div>
        <br />
        <input
          type="button"
          onClick={() => {}}
          className="button cancel modified-btn"
          value={numberOfDraftOrders > 1 ? "Discard All" : "Discard"}
          disabled={isDisabled}
        />
        <input
          type="submit"
          onClick={() => {}}
          className="button confirm right modified-btn"
          value="Sign and Save"
          disabled={isDisabled}
        />
      </div>
    );
  }
}

LabDraftOrder.propTypes = {
  draftLabOrders: PropTypes.arrayOf(PropTypes.any).isRequired,
};

export default LabDraftOrder;
