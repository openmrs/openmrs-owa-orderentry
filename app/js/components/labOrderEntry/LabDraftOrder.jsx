import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import shortid from 'shortid';
import constants from '../../utils/constants';

export class LabDraftOrder extends PureComponent {
  handleToggleDraftOrderUgency = (event, order) => {
    event.preventDefault();
    const orderUuid = order.uuid;
    let orderUrgency;
    const hasUrgencyProperty = Object.prototype.hasOwnProperty.call(order, 'urgency');
    if (hasUrgencyProperty && order.urgency === constants.ROUTINE) {
      orderUrgency = constants.STAT;
    } else if (hasUrgencyProperty && order.urgency === constants.STAT) {
      orderUrgency = constants.ROUTINE;
    } else {
      orderUrgency = constants.STAT;
    }
    const draftOrder = {
      orderUuid,
      orderUrgency,
    };
    this.props.toggleDraftLabOrdersUgency(draftOrder);
  }

  handleDraftItemDiscard = (event, order, draftType) => {
    event.preventDefault();
    this.props.handleDraftDiscard(order, draftType);
  }

  renderDraftList = () => {
    const { draftLabOrders } = this.props;
    return draftLabOrders.map((order) => {
      const isPanel = !!order.set;
      const draftType = !isPanel ? 'single' : 'panel';
      const orderName = order.display;
      const iconClass = classNames(
        'icon-warning-sign scale',
        {
          'i-gray': order.urgency === constants.ROUTINE,
          'i-red': order.urgency === constants.STAT,
        },
      );
      return (
        <li className="draft-list small-font" key={shortid.generate()}>
          <span className="draft-name">{orderName.toLowerCase()}</span>
          <div className="action-btn-wrapper">
            <span className="action-btn">
              <a
                id="draft-toggle-btn" href="#" onClick={event => this.handleToggleDraftOrderUgency(event, order)}
              >
                <i className={iconClass} title="Urgency" />
              </a>
            </span>
            <span className="action-btn right">
              <a
                id="draft-discard-btn" href="#" onClick={event => this.handleDraftItemDiscard(event, order, draftType)}
              >
                <i className="icon-remove scale" title="Discard" />
              </a>
            </span>
          </div>
        </li>);
    });
  }
  render() {
    const { draftLabOrders, handleDraftDiscard, handleSubmit } = this.props;
    const numberOfDraftOrders = draftLabOrders.length;
    const isDisabled = !numberOfDraftOrders;
    return (
      <div className="draft-spacing draft-lab-layout">
        <h5 className="h5-draft-header">
          Unsaved Draft Orders ({numberOfDraftOrders})
        </h5>
        <div className="table-container">
          <ul className="draft-list-container">
            {this.renderDraftList()}
          </ul>
        </div>
        <br />
        <input
          type="button"
          id="draft-discard-all"
          onClick={() => handleDraftDiscard()}
          className="button cancel modified-btn"
          value={numberOfDraftOrders > 1 ? "Discard All" : "Discard"}
          disabled={isDisabled}
        />
        <input
          type="submit"
          onClick={handleSubmit}
          className="button confirm right modified-btn"
          value="Sign and Save"
          disabled={isDisabled}
        />
      </div>
    );
  }
}

LabDraftOrder.propTypes = {
  draftLabOrders: PropTypes.arrayOf(PropTypes.any).isRequired,
  toggleDraftLabOrdersUgency: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleDraftDiscard: PropTypes.func.isRequired,
};

export default LabDraftOrder;
