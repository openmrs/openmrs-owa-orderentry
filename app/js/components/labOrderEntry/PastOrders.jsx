import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

const PastOrders = ({
  orders,
}) => (
  <div>
    <table className="table bordered mw-958-px">
      <thead>
        <tr>
          <th className="w-155-px">Date</th>
          <th>Details</th>
        </tr>
      </thead>
      <tbody>
        {orders.map(order => (
          <tr key={order.uuid}>
            <td>{moment(order.dateActivated).format('DD MMM YYYY')}</td>
            <td>{order.concept.display}</td>
          </tr>
        ))}
      </tbody>
    </table>
  </div>
);

export default PastOrders;

PastOrders.defaultProps = {
  orders: [
    {
      date: '5-May-2018',
      type: 'Hemoglobin Test',
      status: 'awaiting-results',
      id: 1,
    },
  ],
};

PastOrders.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.shape({
    date: PropTypes.string,
    type: PropTypes.string,
    status: PropTypes.string,
    id: PropTypes.number,
  })),
};
