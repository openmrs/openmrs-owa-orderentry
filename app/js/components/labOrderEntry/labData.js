/**
 * This is a data point for the panel and test form field sets,
 * When an API endpoint is provided, it should be replaced with these.
 * * */
export const panelData = [
  {
    id: 1,
    name: 'Hemogram',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
    labCategory: 1,
    urgency: 'routine',
    tests: [
      {
        id: 1,
        test: 'Hemoglobin',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
      {
        id: 3,
        test: 'blood',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
      {
        id: 6,
        test: 'prothrombin',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
    ],
  },
  {
    id: 2,
    name: 'DLC',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
    labCategory: 1,
    urgency: 'routine',
    tests: [
      {
        id: 2,
        test: 'Hematocrit',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
      {
        id: 4,
        test: 'liver',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
      {
        id: 5,
        test: 'sickling',
        urgency: 'routine',
        concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
      },
    ],
  },
];
export const testsData = [
  {
    id: 1,
    test: 'Hemoglobin',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 2,
    test: 'Hematocrit',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 3,
    test: 'blood',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 4,
    test: 'liver',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 5,
    test: 'sickling',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 6,
    test: 'prothrombin',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 7,
    test: 'Electrophore',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
  {
    id: 8,
    test: 'HbA1c',
    urgency: 'routine',
    concept: '6e5f5436-b56f-4c7d-9580-095588024d77',
  },
];
export const labCategories = [
  { id: 1, name: 'blood' },
  { id: 2, name: 'serum' },
  { id: 3, name: 'urine' },
  { id: 4, name: 'semen' },
  { id: 5, name: 'stool' },
  { id: 6, name: 'tissue' },
];
