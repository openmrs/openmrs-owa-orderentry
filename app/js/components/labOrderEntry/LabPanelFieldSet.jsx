import React from 'react';
import PropTypes from 'prop-types';
import { panelData } from './labData';
import '../../../css/grid.scss';

class LabPanelFieldSet extends React.Component {
  render() {
    const { selectedPanelIds, handleTestSelection } = this.props;
    return (
      <fieldset className="fieldset">
        <legend>Panels</legend>
        {
          panelData.map(panel => (
            <button
              id="panel-button"
              className={(selectedPanelIds.includes(panel.id)) ? 'active lab-tests-btn' : 'lab-tests-btn'}
              type="button"
              key={`${panel.id}`}
              onClick={() => handleTestSelection(panel, 'panel')}
            >
              {panel.name}
            </button>
          ))
        }
      </fieldset>
    );
  }
}

LabPanelFieldSet.defaultProps = {
  selectedPanelIds: [],
};

LabPanelFieldSet.propTypes = {
  handleTestSelection: PropTypes.func.isRequired,
  selectedPanelIds: PropTypes.array,
};

export default LabPanelFieldSet;
