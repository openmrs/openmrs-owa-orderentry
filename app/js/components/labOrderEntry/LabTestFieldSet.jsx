import React from 'react';
import PropTypes from 'prop-types';
import { testsData } from './labData';
import '../../../css/grid.scss';

class LabTestFieldSet extends React.Component {
  render() {
    const { selectedTests, selectTest } = this.props;
    const selectedTestIds = selectedTests.map(test => test.id);
    return (
      <fieldset className="fieldset">
        <legend>Tests</legend>
        <div className="flex-row">
          {
            testsData.map((item, index) => (
              <button
                type="button"
                id="category-test-button"
                key={`${item.id}`}
                className={`lab-tests-btn ${(selectedTestIds.includes(item.id)) ? 'active' : ''}`}
                onClick={() => selectTest(item)}
              >
                {item.test}
              </button>
            ))
          }
        </div>
      </fieldset>
    );
  }
}

LabTestFieldSet.propTypes = {
  selectedTests: PropTypes.array.isRequired,
  selectTest: PropTypes.func.isRequired,
};

export default LabTestFieldSet;
