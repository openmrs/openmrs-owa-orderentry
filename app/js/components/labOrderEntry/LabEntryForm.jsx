import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Accordion } from '@openmrs/react-components';
import LabPanelFieldSet from './LabPanelFieldSet';
import LabTestFieldSet from './LabTestFieldSet';
import LabDraftOrder from './LabDraftOrder';
import ActiveOrders from './ActiveOrders';
import PastOrders from './PastOrders';
import createLabOrder from '../../actions/createLabOrder';
import { successToast, errorToast } from '../../utils/toast';
import {
  addTestPanelToDraft,
  removeTestPanelFromDraft,
  removeTestFromDraft,
  addTestToDraft,
  deleteDraftLabOrder,
  toggleDraftLabOrdersUgency,
} from '../../actions/draftLabOrderAction';
import '../../../css/grid.scss';
import './styles.scss';

import fetchLabOrders from '../../actions/labOrders/fetchLabOrders';
import fetchLabConcepts from '../../actions/labOrders/labConceptsAction';

export class LabEntryForm extends PureComponent {
  static propTypes = {
    createLabOrderReducer: PropTypes.shape({
      status: PropTypes.objectOf(PropTypes.bool),
      errorMessage: PropTypes.string,
      labOrderData: PropTypes.object,
    }),
    draftLabOrders: PropTypes.arrayOf(PropTypes.any).isRequired,
    dateFormat: PropTypes.string.isRequired,
    selectedLabPanels: PropTypes.arrayOf(PropTypes.any).isRequired,
    defaultTests: PropTypes.arrayOf(PropTypes.any).isRequired,
    selectedTests: PropTypes.arrayOf(PropTypes.any).isRequired,
    dispatch: PropTypes.func.isRequired,
    labOrders: PropTypes.shape({
      results: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
    }),
    patient: PropTypes.shape({
      uuid: PropTypes.string,
    }),
    encounterType: PropTypes.shape({
      uuid: PropTypes.string,
    }).isRequired,
    encounterRole: PropTypes.shape({
      uuid: PropTypes.string,
    }),
    inpatientCareSetting: PropTypes.shape({
      uuid: PropTypes.string,
    }),
    conceptsAsPanels: PropTypes.array,
    standAloneTests: PropTypes.array,
    session: PropTypes.shape({
      currentProvider: PropTypes.shape({
        person: PropTypes.shape({
          uuid: PropTypes.string,
        }),
        uuid: PropTypes.string,
      }),
      currentLocation: PropTypes.object,
    }),
    orderables: PropTypes.arrayOf(PropTypes.object).isRequired,
    getLabOrderables: PropTypes.string.isRequired,
  };

  static defaultProps = {
    createLabOrderReducer: {
      status: {},
      errorMessage: '',
    },
    encounterRole: {
      uuid: '',
    },
    patient: {
      uuid: '',
    },
    inpatientCareSetting: {
      uuid: '',
    },
    conceptsAsPanels: [],
    standAloneTests: [],
    session: {
      currentProvider: {
        person: {
          uuid: '',
        },
      },
    },
    labOrders: {
      results: [],
    },
  };

  state = {
    categoryUUID: this.props.orderables[0].uuid || 0,
    categoryName: this.props.orderables[0].display,
    selectedPanelIds: [],
    selectedPanelTestIds: [],
  };

  componentDidMount() {
    this.props.dispatch(fetchLabOrders(null, this.props.patient.uuid));
    if (this.state.categoryUUID) {
      this.props.dispatch(fetchLabConcepts(`${this.state.categoryUUID}?v=full`));
    }
  }

  componentWillReceiveProps(nextProps) {
    const {
      selectedLabPanels,
      defaultTests,
    } = nextProps;
    this.setState({
      selectedPanelIds: selectedLabPanels.map(panel => panel.uuid),
      selectedPanelTestIds: defaultTests.map(test => test.uuid),
    });
  }

  componentDidUpdate(prevProps, prevState) {
    const {
      status: { added, error },
      errorMessage,
      labOrderData,
    } = this.props.createLabOrderReducer;
    if (added && labOrderData !== prevProps.createLabOrderReducer.labOrderData) {
      successToast('lab order successfully created');
      this.props.dispatch(fetchLabOrders(null, this.props.patient.uuid));
    }
    if (error) {
      errorToast(errorMessage);
    }
    if (this.state.categoryUUID !== prevState.categoryUUID) {
      this.props.dispatch(fetchLabConcepts(`${this.state.categoryUUID}?v=full`));
    }
  }

  handleTestSelection = (item, type) => {
    const {
      props: {
        draftLabOrders,
        dispatch,
      },
      state: {
        selectedPanelIds,
        selectedPanelTestIds,
      },
    } = this;
    const isSelected = !!draftLabOrders.filter(order => order.uuid === item.uuid).length;

    if ((type === 'panel')) {
      const isSelectedPanel = selectedPanelIds.includes(item.uuid);
      if (isSelectedPanel) {
        dispatch(removeTestPanelFromDraft(item));
      } else {
        dispatch(addTestPanelToDraft(item));
      }
    }
    if (type === 'single') {
      const isSelectedPanelTest = selectedPanelTestIds.includes(item.uuid);
      if (!isSelectedPanelTest && isSelected) dispatch(removeTestFromDraft(item));
      if (!isSelectedPanelTest && !isSelected)dispatch(addTestToDraft(item));
    }
  }

  handleUrgencyChange = (order) => {
    this.props.dispatch(toggleDraftLabOrdersUgency(order));
  }

  discardTestsInDraft = (test, action) => {
    const { dispatch } = this.props;
    if (action === 'single') return dispatch(removeTestFromDraft(test));
    if (action === 'panel') return dispatch(removeTestPanelFromDraft(test));
    return dispatch(deleteDraftLabOrder());
  }

  showFieldSet = () => (
    <div>
      <LabPanelFieldSet
        labCategoryName={this.state.categoryName}
        handleTestSelection={this.handleTestSelection}
        panels={this.props.conceptsAsPanels}
        selectedPanelIds={this.state.selectedPanelIds}
      />
      <LabTestFieldSet
        labCategoryName={this.state.categoryName}
        handleTestSelection={this.handleTestSelection}
        selectedTests={this.props.selectedTests}
        tests={this.props.standAloneTests}
      />
    </div>
  );

  changeLabForm = (id, name) => {
    this.setState({
      categoryUUID: id,
      categoryName: name,
    });
  };

  handleSubmit = () => {
    const { draftLabOrders } = this.props;
    const isEmpty = !draftLabOrders.length;
    if (isEmpty) return;
    const orders = draftLabOrders.map(labOrder => (
      {
        concept: labOrder.uuid,
        careSetting: this.props.inpatientCareSetting.uuid,
        encounter: this.props.encounterType.uuid,
        orderer: this.props.session.currentProvider.uuid,
        patient: this.props.patient.uuid,
        type: 'testorder',
        urgency: labOrder.urgency || 'ROUTINE',
      }
    ));

    const encounterPayload = {
      encounterProviders: [
        {
          encounterRole: this.props.encounterRole.uuid,
          provider: this.props.session.currentProvider.uuid,
        },
      ],
      encounterType: this.props.encounterType.uuid,
      location: this.props.session.currentLocation,
      orders,
      patient: this.props.patient.uuid,
    };
    this.props.dispatch(createLabOrder(encounterPayload));
    this.props.dispatch(deleteDraftLabOrder());
  };

  renderPendingOrders = () => {
    const { labOrderData } = this.props.createLabOrderReducer;
    if (labOrderData.orders) {
      const tests = labOrderData.orders.map(order => order.display);
      return (
        <Accordion open title="Pending Lab Orders">
          <ActiveOrders labOrderData={labOrderData} tests={tests} />
        </Accordion>
      );
    }
    return (
      <Accordion open title="Pending Lab Orders">
        <p>No pending orders</p>
      </Accordion>
    );
  };

  renderPastOrders = () => {
    const { labOrders: { results }, dateFormat } = this.props;
    if (results.length > 0) {
      return (
        <Accordion open title="Past Lab Orders">
          <PastOrders orders={results} dateFormat={dateFormat} />
        </Accordion>
      );
    }
    return (
      <Accordion title="Past Lab Orders">
        <p>No past orders</p>
      </Accordion>
    );
  };

  renderLabDraftOrder = () => (
    <div className="draft-lab-wrapper">
      <LabDraftOrder
        toggleDraftLabOrdersUgency={this.handleUrgencyChange}
        handleDraftDiscard={this.discardTestsInDraft}
        draftLabOrders={this.props.draftLabOrders}
        panelTests={this.state.selectedPanelTestIds}
        handleSubmit={() => this.handleSubmit()}
      />
    </div>
  );

  render() {
    const {
      handleCancel, handleSubmit, renderPendingOrders, renderPastOrders,
      props: { orderables, getLabOrderables },
    } = this;
    return (
      <React.Fragment>
        {
          getLabOrderables ?
            <div className="lab-order-entry">
              <h5>New Lab Order</h5>
              <br />
              <div className="lab-form-wrapper">
                <div className="lab-category">
                  <ul>
                    {orderables.map((orderable, index) => (
                      // eslint-disable-next-line
                      <li key={index}>
                        <a
                          className={this.state.categoryUUID === orderable.uuid ? 'active-category' : ''}
                          href="#"
                          id="category-button"
                          onClick={() => this.changeLabForm(orderable.uuid, orderable.display)}>
                          {orderable.display}
                        </a>
                      </li>
                    ))}
                  </ul>
                  {this.renderLabDraftOrder()}
                </div>
                <div className="order-form-wrapper">
                  <form className="lab-form simple-form-ui">{this.showFieldSet()}</form>
                </div>
              </div>
              <br />
              {renderPendingOrders()}
              <br />
              {(this.props.labOrders.results) && <div>{renderPastOrders()}</div>}
              <br />
            </div>
            :
            <p>No Lab Orderables was found</p>
        }
      </React.Fragment>
    );
  }
}

export const mapStateToProps = ({
  draftReducer: {
    draftLabOrders: {
      orders,
      selectedLabPanels,
      defaultTests,
      selectedTests,
    },
  },
  dateFormatReducer: { dateFormat },
  labConceptsReducer: {
    conceptsAsPanels,
    standAloneTests,
  },
  openmrs: { session },
  fetchLabOrderReducer: { labOrders },
  patientReducer: { patient },
  encounterRoleReducer: { encounterRole },
  encounterReducer: { encounterType },
  careSettingReducer: { inpatientCareSetting },
  createLabOrderReducer,
  labOrderableReducer: { orderables },
  getLabOrderablesReducer: { getLabOrderables },
}) => ({
  draftLabOrders: orders,
  dateFormat,
  conceptsAsPanels,
  standAloneTests,
  selectedLabPanels,
  defaultTests,
  selectedTests,
  labOrders,
  encounterType,
  inpatientCareSetting,
  createLabOrderReducer,
  encounterRole,
  patient,
  session,
  orderables,
  getLabOrderables,
});

export default connect(mapStateToProps)(LabEntryForm);
