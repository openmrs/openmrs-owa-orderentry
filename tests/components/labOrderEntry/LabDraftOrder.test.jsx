import React from 'react';
import { LabDraftOrder } from '../../../app/js/components/labOrderEntry/LabDraftOrder';

let props;
let mountedComponent;
props = {
  draftLabOrders: [
    { id: 6, test: 'prothrombin', urgency: 'routine' }
  ],
  panelTests: [],
  toggleDraftLabOrdersUgency: jest.fn(),
};

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<LabDraftOrder {...props} />);
  }
  return mountedComponent;
};

describe('Component: LabDraftOrder', () => {
  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render on initial setup', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('should toggle draftOrder Urgency', () => {
    const component = getComponent();
    const handleToggleDraftOrderUgency = jest.spyOn(props, 'toggleDraftLabOrdersUgency');
    component.find('#draft-toggle-btn').at(0).simulate('click', {});
    expect(handleToggleDraftOrderUgency).toBeCalled();
  });
});
