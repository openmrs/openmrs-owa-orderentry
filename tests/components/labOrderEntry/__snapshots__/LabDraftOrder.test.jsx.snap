// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LabDraftOrder should render on initial setup 1`] = `
<div
  className="draft-spacing draft-lab-layout"
>
  <h5
    className="h5-draft-header"
  >
    Unsaved Draft Orders (
    1
    )
  </h5>
  <div
    className="table-container"
  >
    <ul
      className="draft-list-container"
    >
      <li
        className="draft-list small-font"
        key="0"
      >
        <span
          className="draft-name"
        >
          prothrombin
        </span>
        <div
          className="action-btn-wrapper"
        >
          <span
            className="action-btn"
          >
            <a
              href="#"
              id="draft-toggle-btn"
              onClick={[Function]}
            >
              <i
                className="icon-warning-sign scale i-gray"
                title="Urgency"
              />
            </a>
          </span>
          <span
            className="action-btn right"
          >
            <a
              href="#"
              id="draft-discard-btn"
              onClick={[Function]}
            >
              <i
                className="icon-remove scale"
                title="Discard"
              />
            </a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <br />
  <input
    className="button cancel modified-btn"
    disabled={false}
    id="draft-discard-all"
    onClick={[Function]}
    type="button"
    value="Discard"
  />
  <input
    className="button confirm right modified-btn"
    disabled={false}
    onClick={[MockFunction]}
    type="submit"
    value="Sign and Save"
  />
</div>
`;
